### PR TITLE
Minor adjustments to Cirrus configuration

### PR DESCRIPTION
**Pull request by stuartmorgan**
- Use a specific version of the plugin tools, in preparation for
  publishing an updated version out of flutter/plugins.
- Use templates the same way we do in flutter/plugins for consistency
  and ease of maintenance.
- Separate some setup steps into their own script steps for better
  inspectability in the UI.
- Enable build tests on `stable` in addition to `master`

## Pre-launch Checklist

- [ ] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [ ] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy].
- [ ] I updated `CHANGELOG.md` to add a description of the change.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

### CODE REVIEW

File: .cirrus.yml [hunk 1]
```diff
@@ -1,5 +1,19 @@
 gcp_credentials: ENCRYPTED[1816835da1e936dabb469b51501856ec8468676e35e967bd0fd720a815498e5ee6c8a6a79219ce273f67bcb8f1aa948a]
 
+tool_setup_template: &TOOL_SETUP_TEMPLATE
+  tool_setup_script:
+    - git fetch origin master # To set FETCH_HEAD for "git merge-base" to work
+    # Pinned version of the plugin tools, to avoid breakage in this repository
+    # when pushing updates from flutter/plugins.
+    - pub global activate flutter_plugin_tools 0.0.45+3
```
Comment (cyanglaz) [inline .cirrus.yml:8]: I didn't get why we need to pin a specific version. Can't the package repo use the new version that to be published? Or you plan to update it to latest version in a separate PR?

**Review by cyanglaz — APPROVED** [review]
LGTM % a minor question

File: .cirrus.yml [hunk 2]
```diff
@@ -1,5 +1,19 @@
...
+
+flutter_upgrade_template: &FLUTTER_UPGRADE_TEMPLATE
+  upgrade_flutter_script:
+    - flutter channel $CHANNEL
```
Comment (cyanglaz) [inline .cirrus.yml:12]: $CHANNEL seems not set.

**Review by cyanglaz — CHANGES_REQUESTED** [review]

File: .cirrus.yml (hunk 1 above)
Comment (stuartmorgan) [reply .cirrus.yml:8]: > Can't the package repo use the new version that to be published? Or you plan to update it to latest version in a separate PR?

Yes and yes. The issue is that we don't have any real concept of "breaking changes" to the tool, at least in the new structure (not sure how it worked before). When we land things we'll only know they work in flutter/plugins, not that they don't break flutter/packages. By pinning the version and then explicitly rolling here, we'll avoid surprise breakage to flutter/plugins from some non-obvious source.

File: .cirrus.yml (hunk 2 above)
Comment (stuartmorgan) [reply .cirrus.yml:15]: Oops, missed a piece. Fixed.

That also made me realize that we're currently only building the packages on master which seems... not great. So I've added stable, like we do in flutter/plugins.

**Review by cyanglaz — APPROVED** [review]
LGTM

**Pull request merged**